### ISSUE TITLE
Tests to make sure the new transform is used

### ISSUE THREAD

**Issue by prakharjoshi** [collective/experimental.safe_html_transform#29]

**Comment by prakharjoshi** [comment]
@tisto  @lentinj  . Can You provide me some guidance how to write tests and to make sure our transform is used instead of old one ?

**Comment by tisto** [comment]
First thing to do would be to write a simple test that passes on your transform but fails on the old safe_html transform. If you have that you can use that test to write a functional end-to-end test (either zope.testbrowser or robotframework).

**Comment by tisto** [comment]
@prakharjoshi what about working on this one next? You could write an acceptance test that takes a reasonably complex HTML input and makes sure all unwanted tags are filtered out. You could test that the transform itself is applied with a zope.testbrowser test (that does not use TinyMCE) and a robot framework based test (that uses TinyMCE).

**Comment by prakharjoshi** [comment]
@tisto  yeah sure lets work on this one and also I have tried to run the script on some complex html pages and it ran successfully on local, now lets write tests which take whole html page as an input and checks the output with the expected output, is that what we want, right ?

**Comment by tisto** [comment]
@prakharjoshi Yes. There is an example robot test in your package:

https://github.com/collective/experimental.safe_html_transform/blob/master/src/experimental/safe_html_transform/tests/robot/test_example.robot

Here is an example of a zope.testbrowser functional test:

https://github.com/plone/plone.app.contenttypes/blob/master/plone/app/contenttypes/tests/test_document.py#L89

**Comment by prakharjoshi** [comment]
@tisto  I am on it. Will set up test for the html pages also :)

**Comment by prakharjoshi** [comment]
@tisto  So we have to create a different file for testing the transform with documents and I didn't get that do we have to write robot tests or normal tests ? As the document test you have provided is a functional test ?

**Comment by lentinj** [comment]
As for something that would fail on the old transform but pass on the new one, remember the conversation about `<img/>` tags on an earlier ticket.

**Comment by lentinj** [comment]
@prakharjoshi Yes, if you just wanted to test a bigger document, you can do that in a unit test like the ones you've already written.

The difference here, and why @tisto is proposing robot tests, is you want to test that the install of the package, and that it's used when editing a TinyMCE (i.e. rich text) field on a page. You can use robotframework to install the add-on exactly as a site administrator might.

**Comment by prakharjoshi** [comment]
@lentinj  Okay yeah I tried to run these tests cases with the SAFEHTML of PortalTransforms and I got 17 test cases failure and when I run these test cases with our new SAFEHTML all test cases are passing successfully :)

**Comment by prakharjoshi** [comment]
@tisto  @lentinj  For testing our new add-on we have to first write tests that just activate our add-on, right ? As when I am trying to write tests then it seems like the test browser on which plone is running have not intsalled our new transform and still `/@@filter-controlpanel` is working and the new url we have provided is not working like `/@@safe_html_transform-filtering` . So Just wanna ask that we have to first instruct the robot framework to first install the new add-on and then we can check for control panel, right ?

**Comment by lentinj** [comment]
Yes, that's the objective here. You need a robot test that first proves the add-on is not active, then activates it, then proves that it is active and used when editing rich text.

**Comment by prakharjoshi** [comment]
@tisto  @lentinj  Okay so when I tired to add a new page to the site while installing our new add-on, I got the error and here are the error logs http://pastie.org/10354463 . As I have seen the logs it seems like the TinyMCE  is still looking for safe_html of PortalTransform and as we have registered the new add-on the old safe_html is unregistered.

**Comment by lentinj** [comment]
Yes, it's looking for a safe_html transform by name, so what you register has to also be called "safe_html".

**Comment by prakharjoshi** [comment]
Okay so I did that in the pull request #39 . I changed the transform name to safe_html and now its working fine. I have added a test for the same. But don't know why travis is not detecting it. :/

**Comment by lentinj** [comment]
Ah okay, I'm getting lost in the various threads.

However, looks like tisto sorted you out on the Travis front? Your last commit was tested and passed,
